dev/examples/simulation/double_ellipsoid.clj

```diff
@@ -2,70 +2,85 @@
-            [nextjournal.clerk.viewer :as-alias viewer]))
+            [emmy.mathbox.physics]
+            [emmy.viewer.physics]
+            [nextjournal.clerk :as clerk]))
+
+{::clerk/width :wide}
+
+^{::clerk/visibility {:code :hide :result :hide}}
+(ec/install!)
 
 ;; ## Ellipsoid with Two
 
-(defn elliptical->rect [a b c]
-  (fn [[_ [theta phi theta2 phi2] _]]
-    (up (* a (sin theta) (cos phi))
-        (* b (sin theta) (sin phi))
-        (* c (cos theta))
-        (* a (sin theta2) (cos phi2))
-        (* b (sin theta2) (sin phi2))
-        (* c (cos theta2)))))
+(defn elliptical->rect* [a b c]
+  (let [xform (map (ell/elliptical->rect a b c))]
+    (partial into (up) xform)))
+
+(defn e->r* [a b c]
+  (comp (elliptical->rect* a b c)
+        coordinate))
 
 ;; Next, the Lagrangian:
 
-(defn L-central-triaxial [m a b c]
+(defn U [k x0]
+  (fn [[_ [x1 x2]]]
+    (let [diff (abs (- x2 x1))]
+      (* 1/2 k (square (- diff x0))))))
+
+(defn L-central-triaxial [m k x0 a b c]
   (comp (- (ell/L-free-particle m)
-           ;; POTENTIAL ENERGY TERM!!!!
-           (fn [[_ [x1 y1 z1 x2 y2 z2]]]
-             (let [k  10
-                   x0 0.3
-                   x  (abs
-                       (- [x1 y1 z1]
-                          [x2 y2 z2]))]
-               (* (/ 1 2)
-                  k
-                  (square (- x x0))))))
-        (F->C (elliptical->rect a b c))))
+           (U k x0))
+        (F->C (e->r* a b c))))
 
 ;; Final Demo:
-#_
-(clerk/with-viewer
-  {:transform-fn pv/physics-xform
-   :render-fn
-   (q
-    (fn [value]
-      (viewer/html
-       [mathbox/MathBox ~pv/opts
-        [mb/Cartesian (:cartesian value)
-         [mb/Axis {:axis 1 :width 3}]
-         [mb/Axis {:axis 2 :width 3}]
-         [mb/Axis {:axis 3 :width 3}]
-         [examples.simulation.utils/Ellipse (:ellipse value)]
-         [examples.simulation.utils/DoubleMass
-          (select-keys value [:L :state->xyz :initial-state])]]])))}
 
-  (let [m 10 a 2 b 1 c 1]
-    {:state->xyz (elliptical->rect a b c)
-     :L          (L-central-triaxial m a b c)
-     :initial-state [0 [0.1 0.1 2 2]
-                     [0.3 0.3 0 0]]
-     :ellipse {:a a :b b :c c}
-     :cartesian
-     {:range [[-10 10]
-              [-10 10]
-              [-10 10]]
-      :scale [3 3 3]}}))
+(let [initial-state [0
+                     [[0.1 0.1] [2 2]]
+                     [[0.3 0.3] [0 0]]]]
+  (ev/with-let   [!state {:time 0 :state initial-state}]
+    (ev/with-let [!opts {:m 10 :a 2 :b 1 :c 1 :k 10 :x0 0.3}]
+      (plot/scene
+       {:threestrap {:plugins ["core" "controls" "cursor" "stats"]}}
+       (leva/controls {:atom !opts})
+
+       (emmy.viewer.physics/evolve
+        {:atom !state
+         :initial-state initial-state
+         :f' (ev/with-params {:atom !opts :params [:m :k :x0 :a :b :c]}
+               (comp Lagrangian->state-derivative
+                     L-central-triaxial))})
+
+       (emmy.mathbox.physics/comet
```

Comment:
it would be nice to be able to emit multiple items. @littleredcomputer what are your thoughts on a compiled function convention if we want to emit multiple items from a single return value?